packages/gatsby-theme-carbon/src/util/hooks/useNavScroll.js

```diff
@@ -0,0 +1,16 @@
+import { useEffect, useContext } from 'react';
+import NavContext from '../context/NavContext';
+
+const useNavScroll = () => {
+  const { updateNavScrollOffset } = useContext(NavContext);
+
+  useEffect(() => {
+    const sideNav = document.querySelector('ul.bx--side-nav__items');
```

Comment:
Could we instead have the hook take a ref? in general doing query selectors in react can get hairy really quick.